CONTRIBUTING.md

```diff
@@ -1,24 +1,56 @@
-- [ ] Did you run lint (`yarn lint:check`) and fix any issues? (`yarn lint:fix`)
-- [ ] Is the base branch you're merging into `development` and not `main`?
+2. **Install Dependencies**: Ensure you have `yarn` installed (we use version 1.x). Installation instructions are available [here](https://classic.yarnpkg.com/lang/en/docs/install). Run `yarn` in the project root to install all necessary dependencies.
 
-## Docs guidelines
+3. **Testing Environment**: Before making changes, make sure to verify the testing environment.
 
-We leverage `vitepress` to build our docs. You can find the docs in the `site` folder. To run the docs locally, run `yarn docs:dev` from the `site` folder. To build the docs, run `yarn docs:build`. Docs are automatically deployed with each package published.
+   - Use the Node version specified in `package.json` (currently 18.10). Run `node -v` to check your version.
+   - Build the project with `yarn build`.
+   - Run existing tests using `yarn test` to ensure everything is working correctly.
 
-When updating code, be sure to update the relevant doc within `site/packages/*`. The `packages` directory is broken down by the relevant package found within this repository. When adding new functionality, be sure to add a new doc outlining the method or class that you've added!
+4. **Make Changes**: Now, you can start making changes to the packages or docs. When updating or adding new functionality, update or add a new doc in `site/packages/*` corresponding to the package you've worked on to document the changes.
+
+5. **Reverify Tests**: After making your changes, re-run `yarn test` to ensure all tests still pass.
+
+6. **Code Formatting**:
+
+   - Format your code changes with `yarn run lint:write`.
+   - Confirm that your code passes format checks with `yarn run lint:check`.
+
+7. **Docs Changes**:
+
+   - We use `vitepress` for our docs located in the `site` folder.
+   - To run docs locally: `yarn dev` from the `site` folder.
+   - To build docs: `yarn build` from the `site` folder.
+   - When editing or adding new docs, make sure you follow the guidelines mentioned below:
+     - Follow the [Google style guidelines](https://developers.google.com/style) for docs content.
+     - Additional Guidelines:
+       - Use terms consistently (e.g., "smart account", not "Smart Account").
+       - Refer to "Account Kit" correctly, without "the" or "AccountKit".
+       - Use `LightAccount` for code references, "Light Account" in text.
+       - Use "gasless" over "gas-less".
+       - Write documentation in the [second person voice](https://developers.google.com/style/person).
+       - Use "aa-sdk" or "Account Kit" depending on context, not "Account Kit SDK".
+       - Capitalize "Gas Manager API" and "Bundler API".
+
+8. **Committing Changes**: Commit your changes using a standardized message format.
```

Comment:
link to something demonstrating semantic pr titles, like this: https://flank.github.io/flank/pr_titles/